padinfo/boardgenerator.py

```diff
@@ -0,0 +1,88 @@
+class BoardGenerator(object):
+    """Generate a Dawnglare link with optional inversion."""
+
+    allowed_letters = "RGBLDHXJMPZ "
+    inversion = "-1"
+    dawnglare_link = "https://pad.dawnglare.com/?patt={}&showfill=1"
+
+    def __init__(self, board):
+        self.board = board.split()
+        self.invalid_board = False
+        self.invert = self.inversion_check()
+        self.generate_clean_board()
+        self.validate_orbs()
+        self.board_length = len(self.clean_board)
+        self.eval_board_size()
+        self.generate_link()
+
+    def inversion_check(self):
+        if BoardGenerator.inversion in self.board:
+            invert = True
+        else:
+            invert = False
+
+        return invert
+
+    def generate_clean_board(self):
```

Comment:
We generally want to restrict our function names to use a set of a few verbs.  Some that I like are `get`, `set`, and `validate` which I think are super clear!